javascript-challenges.test.js

```diff
@@ -14,30 +16,97 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("codeWord" , () => {
+    it("takes in a string and returns a string with a coded message", () => {
+        expect(codeWord(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codeWord(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codeWord(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
-// b) Create the function that makes the test pass.
+// ReferenceError: codeWord is not defined
 
-// Pseudo code:
+const secretCodeWord1 = "Lackadaisical";
+// // Expected output: "L4ck4d41s1c4l"
+const secretCodeWord2 = "Gobbledygook";
+// // Expected output: "G0bbl3dyg00k"
+const secretCodeWord3 = "Eccentric";
+// // Expected output: "3cc3ntr1c"
+
+// // b) Create the function that makes the test pass.
+
+// // Pseudo code:
+// // created a function codeWord with a parameter string
+// // created a temporary variable array = string.split("") to turn the string into an array and seperate each index
+// // use a for loop to iterate through the array then use the conditionals (if and if else) to locate a specific index and swap the character("aeio") with an interger (4,3,1,0).
+// // return the function and the temporary variable array. use .join("") to switch the array into the string
+
+const codeWord = (string) => {
+  let array = string.split("");
+  for (let i = 0; i < array.length; i++) {
+     if (array[i] === "a" || array[i] === "A") {
+      array[i] = 4
+    }
+     else if (array[i] === "e" || array[i] === "E" ) {
+        array[i] = 3
+    }
+    else if (array[i] === "i" || array[i] === "I") {
+        array[i] = 1
+    }
+    else if (array[i] === "o" || array[i] === "O") {
+        array[i] = 0
+    }
+  }
+  return array.join("")
+}
+
```

Comment:
Awesome!  Way to stick to the basics and get the job done! One note, you don't need to console.log your function calls because you should be relying on the tests!